test/common/http/http2/codec_impl_test.cc

```diff
@@ -188,6 +188,25 @@ TEST_P(Http2CodecImplTest, InvalidRepeatContinue) {
   EXPECT_THROW(response_encoder_->encodeHeaders(continue_headers, true), CodecProtocolException);
 };
 
+TEST_P(Http2CodecImplTest, Invalid103) {
+  initialize();
+
+  TestHeaderMapImpl request_headers;
+  HttpTestUtility::addDefaultHeaders(request_headers);
+  EXPECT_CALL(request_decoder_, decodeHeaders_(_, true));
+  request_encoder_->encodeHeaders(request_headers, true);
+
+  TestHeaderMapImpl continue_headers{{":status", "100"}};
+  EXPECT_CALL(response_decoder_, decode100ContinueHeaders_(_));
+  response_encoder_->encode100ContinueHeaders(continue_headers);
+
+  TestHeaderMapImpl early_hint_headers{{":status", "103"}};
+  EXPECT_CALL(response_decoder_, decodeHeaders_(_, false));
+  response_encoder_->encodeHeaders(early_hint_headers, false);
+
+  EXPECT_THROW(response_encoder_->encodeHeaders(early_hint_headers, false), CodecProtocolException);
```

Comment:
nit: It's a little hard to tell whether this is throwing when encodeHeaders is called, or when we process on the other side. Is it possible to disable byte transfer temporarily and then check for an exception when we parse bytes on the other side?

Reply:
Switching to EXPECT_THROW_WITH_MESSAGE which I think makes it clear where the exception is coming from.  WDYT?